lib.php

```diff
@@ -980,3 +999,31 @@ function reengagement_checkstart($course, $cm, $reengagement) {
     }
     return $output;
 }
+
+/**
+ * Check that emailfrom user has capability to add reengagments,
+ *  otherwise return support user
+ *
+ * @param object $reengagement
+ * @return object $user
+ */
+function reengagement_get_emailfrom($reengagement) {
+    $userid = $reengagement->emailfrom;
+    if ($userid > 0) {
+        $context = context_course::instance($reengagement->course);
+        if ($userid == 1) { // Default teacher, get first teacher in course.
+            global $DB;
+            $params = array('roleid' => 3, 'contextid' => $context->id);
```

Comment:
hard-coded roleid - this should be a site-level setting for re-engagement that allows the admin to select a role that is used for teaching staff or better still, use a has_capability check instead.

Reply:
in fact - it would be better to use the get_course_contacts() style handling here with the $CFG->coursecontact settings instead of hard-coding this to a specific role.

Reply:
changed this to get_enrolled_users() with a permissions check for 'mod/reengagement:addinstance'